activiti-cloud-service-common/activiti-cloud-services-swagger/src/main/java/springfox/documentation/schema/TypeNameExtractor.java

```diff
@@ -0,0 +1,127 @@
+/*
+ *
+ *  Copyright 2015 the original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ */
+
+package springfox.documentation.schema;
+
+import static com.google.common.base.Optional.fromNullable;
+import static springfox.documentation.schema.Collections.containerType;
+import static springfox.documentation.schema.Collections.isContainerType;
+import static springfox.documentation.schema.Types.typeNameFor;
+
+import com.fasterxml.classmate.ResolvedType;
+import com.fasterxml.classmate.TypeResolver;
+import com.fasterxml.classmate.types.ResolvedArrayType;
+import com.fasterxml.classmate.types.ResolvedObjectType;
+import com.fasterxml.classmate.types.ResolvedPrimitiveType;
+import java.lang.reflect.Type;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.plugin.core.PluginRegistry;
+import org.springframework.stereotype.Component;
+import springfox.documentation.schema.DefaultTypeNameProvider;
+import springfox.documentation.schema.ModelNameContext;
+import springfox.documentation.spi.DocumentationType;
+import springfox.documentation.spi.schema.EnumTypeDeterminer;
+import springfox.documentation.spi.schema.GenericTypeNamingStrategy;
+import springfox.documentation.spi.schema.TypeNameProviderPlugin;
+import springfox.documentation.spi.schema.contexts.ModelContext;
+
+@Component
```

Comment:
Why do we need this class here? Are we patching something from former `TypeNameExtractor`?

Reply:
using new spring-plugin API which are incompatible

Reply:
I missed this comment https://github.com/Activiti/Activiti/issues/3019#issuecomment-560478416